icu4c/source/test/intltest/plurults.cpp

```diff
@@ -595,6 +597,80 @@ PluralRulesTest::testGetAllKeywordValues() {
     }
 }
 
+void
+PluralRulesTest::testCompactDecimalPluralKeyword() {
+    IcuTestErrorCode errorCode(*this, "testCompactDecimalPluralKeyword");
+
+    LocalPointer<PluralRules> rules(PluralRules::createRules(
+        "one: i = 0,1 @integer 0, 1 @decimal 0.0~1.5;  many: e = 0 and i % 1000000 = 0 and v = 0 or e != 0 .. 5;  other:  @integer 2~17, 100, 1000, 10000, 100000, 1000000, @decimal 2.0~3.5, 10.0, 100.0, 1000.0, 10000.0, 100000.0, 1000000.0, …", errorCode));
+
+    const char* localeName = "fr-FR";
+    Locale locale = Locale::createFromName(localeName);
+
+    struct TestCase {
+        const char* skeleton;
```

Comment:
Optional: It would be more correct to make this `char16_t*` (and `u""` strings), because number skeletons are 16-bit strings.  Currently you are performing an implicit codepage conversion in the `NumberFormatter:forSkeleton` function.

Reply:
Done.